### ISSUE TITLE
Cannot enable prod mode after platform setup

### ISSUE THREAD

**Issue by gnujeremie** [angular/angular#10643]
**I'm submitting a ...**
```
[x] bug report
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
Calling "enableProdMode()" in main.ts raises an error : "Cannot enable prod mode after platform setup"

**Expected/desired behavior**
No error, (as in RC4), and prod mode enable.

**Reproduction of the problem**
See for example the main.ts from the document https://angular.io/docs/ts/latest/guide/webpack.html#!#try (with webpack)

**Please tell us about your environment:**

* **Angular version:** 2.0.0-rc.5
 
* **Browser:** [ Firefox ] 
 
* **Language:** [TypeScript ] 

Webpack used instead of system.js.

**Comment by pkozlowski-opensource** [comment]
This will be solved by #12200

**Comment by vprasanth** [comment]
Is there a workaround for this? Why is this closed, this should get closed when #12200 is closed. I don't want to hijack that issue for this type of conversation.
